pinecone/core/grpc/index_grpc.py

```diff
@@ -251,11 +245,51 @@ def traceback(self, timeout=None):
 
 class GRPCIndex(GRPCIndexBase):
 
+    """A client for interacting with a Pinecone index via GRPC API."""
+
     @property
     def stub_class(self):
         return VectorServiceStub
 
-    def upsert(self, vectors, async_req=False, **kwargs):
+    def upsert(self,
+               vectors: Union[List[GRPCVector], List[Tuple]],
```

Comment:
You need to add `Vector` from `pinecone.core.model.vector` to the union

Reply:
I don't think it's possible to pass this vector, only the grpc one

Reply:
Oh, wait, this is `GRPCIndex` 🤦 .  
You're right, I was thinking about the OpenAPI index. My bad